src/peergos/server/net/SocialHandler.java

```diff
@@ -0,0 +1,84 @@
+package peergos.server.net;
+
+import com.sun.net.httpserver.*;
+import peergos.server.*;
+import peergos.server.social.*;
+import peergos.server.util.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.social.*;
+import peergos.shared.util.*;
+
+import java.io.*;
+import java.util.*;
+import java.util.function.*;
+import java.util.logging.*;
+
+public class SocialHandler implements HttpHandler {
+    private static final Logger LOG = Logging.LOG();
+
+    private final SocialNetwork social;
+
+    public SocialHandler(SocialNetwork social) {
+        this.social = social;
+    }
+
+    public void handle(HttpExchange exchange) throws IOException
+    {
+        long t1 = System.currentTimeMillis();
+        DataInputStream din = new DataInputStream(exchange.getRequestBody());
+
+        ByteArrayOutputStream bout = new ByteArrayOutputStream();
+        DataOutputStream dout = new DataOutputStream(bout);
+
+        String path = exchange.getRequestURI().getPath();
+        if (path.startsWith("/"))
+            path = path.substring(1);
+        String[] subComponents = path.substring(UserService.SOCIAL_URL.length()).split("/");
+        String method = subComponents[0];
+        Map<String, List<String>> params = HttpUtil.parseQuery(exchange.getRequestURI().getQuery());
+        Function<String, String> last = key -> params.get(key).get(params.get(key).size() - 1);
```

Comment:
why not get the first  element?

Reply:
Later duplicate parameters override earlier ones.